hudi-common/src/test/java/org/apache/hudi/common/util/collection/TestExternalSpillableMap.java

```diff
@@ -339,7 +343,42 @@ public void testDataCorrectnessWithoutHoodieMetadata(ExternalSpillableMap.DiskMa
 
   // TODO : come up with a performance eval test for spillableMap
   @Test
-  public void testLargeInsertUpsert() {}
+  public void testLargeInsertUpsert() {
+  }
+
+  @Test
+  public void testPayloadSizeEstimate() throws IOException, URISyntaxException {
+    final ExternalSpillableMap.DiskMapType diskMapType = ExternalSpillableMap.DiskMapType.BITCASK;
+    final boolean isCompressionEnabled = false;
+    final Schema schema = SchemaTestUtil.getSimpleSchema();
+
+    ExternalSpillableMap<String, HoodieRecord<? extends HoodieRecordPayload>> records =
+        new ExternalSpillableMap<>(16L, basePath, new DefaultSizeEstimator(),
+            new HoodieRecordSizeEstimator(schema), diskMapType, isCompressionEnabled);
+
+    List<String> recordKeys = new ArrayList<>();
+
+    // Put a single record. Payload size estimation happens as part of this initial put.
+    HoodieRecord seedRecord = SchemaTestUtil.generateHoodieTestRecordsWithoutHoodieMetadata(0, 1).get(0);
+    records.put(seedRecord.getRecordKey(), seedRecord);
+
+    // Remove the key immediately to make the map empty again.
+    records.remove(seedRecord.getRecordKey());
+
+    // Payload size re-estimation should not happen as the map
+    // size has not reached the minimum size threshold for
+    // recalculation.
+    records.put(seedRecord.getRecordKey(), seedRecord);
+
+    // Put more records than the threshold to trigger payload size re-estimation
+    while (records.getDiskBasedMapNumEntries() < 1) {
```

Comment:
what is this test asserting?  I don't see any conditions being tested?

Reply:
Without the fix, Map put operation throws exception and fails the test. To make it more clear on the expectation of the test, added the `assertDoesNotThrow` to the line where the exception will be thrown without the fix.

Reply:
do we need the lines 371-379. they are just trying to validate no error is thrown? in general, tests should have clear asserts. My suggestion would be to scope down this test to just testing the divide by zero scenario.

Reply:
Removed the while block. But, I still need to put more than 100 entries to verify the payload size re-estimation code path. Updated the test. Please take one more look.